plugins/out_es/es_conf.c

```diff
@@ -153,8 +157,41 @@ struct flb_elasticsearch *flb_es_conf_create(struct flb_output_instance *ins,
             flb_es_conf_destroy(ctx);
             return NULL;
         }
+        flb_plg_debug(ctx->ins, "extracted cloud_host: '%s'", cloud_host);
+
+        flb_plg_debug(ctx->ins,
+                      "check whether extracted cloud_host '%s' contains a port",
+                      cloud_host);
+        cloud_host = strtok_r(cloud_host, ":", &saveptr);
```

Comment:
You need to save `cloud_host` before calling `strtok_r` because it's dynamically allocated by `extract_cloud_host` and if the `strtok_r` call fails we need to release it.

Reply:
Could you give an example?

Reply:
```
char *cloud_host_colon_ptr;

cloud_host_colon_ptr = strtok_r(cloud_host, ":", &saveptr);

if (cloud_host_colon_ptr == NULL) {
    [...]
    flb_plg_error
    flb_sds_destroy(cloud_host);
    [...]
}
```

Something like that except you would have to define the variable in line 135 and be sure to destroy it in the exit path that ends in line 170 as well.

Reply:
Resolved by using strchr.